src/settings/storeconf.cpp

```diff
@@ -9,126 +9,165 @@
 #include <QFileDialog>
 #include <QApplication>
 
-StoreConf::StoreConf(QWidget * parent)
-    : QWidget (parent)
+StoreConf::StoreConf(QString userName)
+    : mConFile {userName += ".json"}
 {
-    conFile = getPathToConFile();
-    if (!conFile.size()) // conf.json file not exist. keep all fields with default values,
-    {                   // create conf.json with these values
+    mFields["applicationName"] = "Pair Storm";       //  default values used when json file
+    mFields["applicationVersion"] = "0.1";           //      not exist or corrupted
+    mFields["organizationName"] = "Lv-425.C++";
+    mFields["style"] = "WHITE";
+    mFields["editorFontName"] = "Consolas";
+    mFields["editorFontSize"] = "12";
+    mFields["defaultUdpPortNumber"] = "36108";
+    mFields["defaultTcpPortNumber"] = "32807";
+    mFields["userName"] = "unnamed user";
+    mFields["cppExtentions"] = ".c;.cpp;.h;.hpp;.json;.txt";
+    mCppExtentionsList << ".c" << ".cpp" << ".h" << ".hpp" << ".json" << ".txt";
+
+}
+
+void StoreConf::restoreConFile()
+{
+    checkConfDir();
+    getPathToConFile();
+    QFile file(mPathToConFile);
+    if (!file.exists()) // userName.json file not exist. keep all fields with default values,
+    {                   // create userName.json with these values
         writeJson();
         saveData();
-    } else {
+    }
+    else
+    {
         readJson();
-        parseJson();
+        if (mReadStatus)
+        {
+            parseJson();
+        }
+        else
+        {                        //  file corrupted
+            writeJson();        //      rewrite with default values
+        }
         saveData();
     }
 }
 
-QString StoreConf::getPathToConFile()
+void StoreConf::saveConFile()
 {
-    QString currentPath = QDir::currentPath();
-    currentPath += "/conf.json";
-    QFile file(currentPath);
-    if(file.exists())
-        return "conf.json";
-    else
-        return "";
+    getPathToConFile();
+    QFile file(mPathToConFile);
+    if (file.exists())
+    {
+        writeJson("finish");
+    }
+}
+
+void StoreConf::getPathToConFile()
+{
+    mPathToConFile = QDir::currentPath();
+    mPathToConFile += "/conf/";
+    mPathToConFile += mConFile;
+}
+
+void StoreConf::checkConfDir(QString str)
+{
+    if (str == "exe")
+    {
```

Comment:
current method is called only once in code (line 31) & is not passed any arg
Therefore default param will be used in 100% cases
Than this if statement will evaluate to true in any case

Reply:
to get posibility to change location directory for configuration files signature of checkConDir has argument which passes this information
by default it lives in same directory as execution file: so parameter = "exe"